web/src/components/watch.tsx

```diff
@@ -37,12 +42,64 @@ export default function Watch(props: { name: string }) {
 
 	const play = () => usePlayer()?.play()
 
+	const toggleCatalog = () => {
+	    setShowCatalog(prev => {
+		const newState = !prev;
+		const catalogBox = document.getElementById("catalogBox");
+
+		if (catalogBox) {
+			catalogBox.style.display = newState ? 'block' : 'none';
```

Comment:
You should use <Show> for this instead. https://www.solidjs.com/tutorial/flow_show